### ISSUE TITLE
macOS / iOS Calendar.app client support?

### ISSUE THREAD

**Issue by seanm** [Kozea/Radicale#1217]
More of a question than a bug report, but... the list of supported clients here:

https://radicale.org/v3.html#supported-clients

does not include the macOS / iOS Calendar applications.  Is this an oversight, or are they indeed not supported?

Thanks.

**Comment by alqu** [comment]
Hm, I am running radicale ONLY with MacOS and iOS clients. Works flawlessly.

Though there is one caveat: you have to import the SSL-Certificate of your Radicale-Server manually to your iOS / Mac beforehand to make it work (only if you created the certificate by yourself).

I am running Radicale behind an NginX-proxy.

**Comment by seanm** [comment]
Thanks for that info!  So I guess this is a bug report afterall, against the documentation.

I note that the [baikal](https://sabre.io/baikal/) docs say macOS is supported... It almost caused me to discard radicale...

**Comment by fredthomsen** [comment]
@alqu I have been running radicale v2 for sometime and tried to get macOS working with v2 and a new instance of v3 I spun up. I've spun this up on my local network as a test, no proxying or SSL is enabled.  Neither seem to work rather I just get "Unable to verify account name or password".  Logs seem to indicate the server is not getting hit with every request to login which I am confused by.  Is there some simple trick I am missing?

**Comment by alqu** [comment]
Yes, there is a trick. To make it actually work on your own server with your self signed certs, you need to manually import your CA-cert into the keystore. And then trust this cert.

Result: All certs that are created under that CA are automatically trusted - and then automagically the login works.

I did this with the CA's certs since IOS and/or MacOS trusts all certs that are issued by this CA. :-)

**Comment by alqu** [comment]
...and you wrote "no SSL" - this could be also the culprit since this is a security measure by IOS to NOT trust unencrypted connections. Yeah. I took me a while to find that out.

**Comment by fredthomsen** [comment]
Ah ok. Now I see a reference to this here as well: https://github.com/Kozea/Radicale/issues/967#issuecomment-511547885 explicitly mentioning macOS.  Thanks for the info.

**Comment by lingdocs** [comment]
> Hm, I am running radicale ONLY with MacOS and iOS clients. Works flawlessly.
> 
> Though there is one caveat: you have to import the SSL-Certificate of your Radicale-Server manually to your iOS / Mac beforehand to make it work (only if you created the certificate by yourself).
> 
> I am running Radicale behind an NginX-proxy.

@alqu Will this work with a Letsencrypt certificate?

**Comment by alqu** [comment]
> @alqu Will this work with a Letsencrypt certificate?

I think it would work **even better** with Letsencrypt since it is a valid one right out of the box... but have not tested by myself.

**Comment by GaiusVampus** [comment]
First we followed the instructions and examples provided in the Radicale documentation which recommended the Location /radicale. After the initial installation seemed to work we later found out the error regarding the MOVE of an item from one calendar to another in macOS/iOS clients. Then to eliminate errors we simplified the setup to the following:

```
<Location "/">
    AuthType     Basic
    AuthName     "Password Required"
    AuthUserFile "/etc/radicale/users"
    Require      valid-user
    ProxyPass        http://localhost:5232/ retry=0
    ProxyPassReverse http://localhost:5232/
    RequestHeader    set X-Remote-User expr=%{REMOTE_USER}
    RequestHeader    set X-Real-IP expr=%{REMOTE_ADDR}
    RequestHeader    set X-Forwarded-Proto https
</Location>
```

Again, almost everything works fine but macOS/iOS clients use MOVE instead of DELETE/CREATE which e.g. Thunderbird uses for that purpose. Hence, Radicale has to process the URL passed by the client in order to reference the destination calendar – which fails (interesting parts of the debug logfile):

```
'REQUEST_METHOD': 'MOVE',
…
Base prefix (from SCRIPT_NAME): ''
…
Sanitized path: '/_username_/_calendarID_.ics
…
Unsupported destination address: 'https://_host_._domain_._TLD_/_username_/_calendarID_.ics'
Response content: Remote destination not supported.
MOVE response status for '/_username_/_calendarID.ics' in 0.003 seconds: 502 Bad Gateway
```
The destination URL looks correct and should be resolvable and accessible from Radicale. The error code 'Bad Gateway' doesn't suggest a problem with user credentials either. 

Thanks for having a look! Just let us know if there is anything else we could test or any further information to provide in order to help this to be fixed.

**Comment by AvdN** [comment]
@GaiusVampus thanks for that confirmation. I added another DNS entry  and made the Apache entry for 443 and that vhost name to redirect to the docker container. 
After that I still had some errors, which had to do with one calendar with old items not having write permissions for the Radicale process (that was not used on the normal clients that use a full URL and no detection, macOS finds it and tries it anyway).

After that everything worked with Apple Calendar, so it certainly does not look like a good idea to rewrite a subpath if working with macOS clients.

**Comment by GaiusVampus** [comment]
Radicale with the PR in #1271 seems to be fully compatible with iOS/macOS devices.

**Comment by pbiering** [comment]
> Radicale with the PR in #1271 seems to be fully compatible with iOS/macOS devices.

Have overseen that particular ticket until today, but got hit by the issue by myself today and during writing a new ticket I found how to solve it. Thank you for your confirmation.
